modules/core/src/main/java/org/apache/ignite/IgniteSystemProperties.java

```diff
@@ -1394,6 +1395,14 @@
      */
     public static final String IGNITE_INDEX_COST_FUNCTION = "IGNITE_INDEX_COST_FUNCTION";
 
+    /**
+     * Enables setting attribute value of {@link
+     * TcpCommunicationSpi#ATTR_HOST_NAMES ATTR_HOST_NAMES} when value {@link
+     * IgniteConfiguration#getLocalHost getLocalHost} is ip.
+     * By default, {@code false}.
```

Comment:
It's not clear from javadoc why do we need this property. Can you please expand it?
As I understand, we want to change current behavior to following one -  ignite always sends ATTR_ADDRS and if the user configures local address as hostname, ignite additionally send ATTR_HOST_NAMES. But we also want to keep backward compatibility(when ignite always send ATTR_HOST_NAMES), so this property added exactly for this purpose, am I right?